server/src/main/java/org/elasticsearch/common/geo/Extent.java

```diff
@@ -115,29 +115,31 @@ static void readFromCompressed(ByteArrayDataInput input, Extent extent) {
                 posRight = Integer.MIN_VALUE;
                 break;
             case POSITIVE_SET:
-                posRight = input.readVInt();
-                posLeft =  Math.toIntExact(posRight - input.readVLong());
+                posLeft = input.readVInt();
+                posRight =  Math.toIntExact(input.readVLong() + posLeft);
                 negLeft = Integer.MAX_VALUE;
                 negRight = Integer.MIN_VALUE;
                 break;
             case NEGATIVE_SET:
-                negRight = input.readInt();
+                negRight = -input.readVInt();
                 negLeft = Math.toIntExact(negRight - input.readVLong());
                 posLeft = Integer.MAX_VALUE;
                 posRight = Integer.MIN_VALUE;
                 break;
             case CROSSES_LAT_AXIS:
-                posRight = input.readInt();
+                posRight = input.readVInt();
                 negLeft = Math.toIntExact(posRight - input.readVLong());
                 posLeft = 0;
                 negRight = 0;
                 break;
-            default:
+            case ALL_SET:
                 posRight = input.readVInt();
                 posLeft =  Math.toIntExact(posRight - input.readVLong());
                 negRight = input.readInt();
                 negLeft = Math.toIntExact(negRight - input.readVLong());
```

Comment:
we should probably update `ALL_SET` to have the same logic as `POSITIVE_SET` and `NEGATIVE_SET` combined?

Reply:
updated to what I think you meant

Reply:
Actually I was thinking of doing this:

```
                posLeft = input.readVInt();
                posRight =  Math.toIntExact(input.readVLong() + posLeft);
                negRight = -input.readVInt();
                negLeft = Math.toIntExact(negRight - input.readVLong());
```

There is no reason to serialize posLeft/posRight differently in the `POSITIVE_SET` case or in the `ALL_SET` case? And likewise for negLeft/negRight in the `NEGATIVE_SET` or `ALL_SET` cases?

Reply:
my bad. I misunderstood. will update thanks